chalicelib/checks/deployment_checks.py

```diff
@@ -0,0 +1,189 @@
+        return check
+
+    def _deploy_indexer(e, version):
+        if is_fourfront_env(e):
+            description = try_to_describe_indexer_env(FF_ENV_INDEXER)
+        else:
+            description = try_to_describe_indexer_env(CGAP_ENV_INDEXER)
+        if description is not None:
+            check.status = 'ERROR'
+            check.summary = 'Tried to spin up indexer env for %s when one already exists for this portal' % e
+            return False
+        else:
+            return EBDeployer.deploy_indexer(e, version)
+
+    if is_cgap_env(env) or is_fourfront_env(env):
+        success = _deploy_indexer(env, application_version)
+        if success:
+            check.status = 'PASS'
+            check.summary = 'Successfully triggered indexer-server provision for environment %s' % env
+        else:
+            check.status = 'ERROR'
+            check.summary = 'An error occurred on deployment. Check the AWS EB Console.'
+    else:
+        check.status = 'FAIL'
+        check.summary = 'Gave an unknown environment: %s' % env
+
+    return check
+
+
+@check_function(env='fourfront-mastertest',
+                branch='master',
+                application_version_name=None, repo=None)
+def deploy_application_to_beanstalk(connection, **kwargs):
+    """ Deploys application to beanstalk under the given application_version name + a branch
+
+        NOTE: CGAP requires kwargs!
+    """
+    check = CheckResult(connection, 'deploy_application_to_beanstalk')
+    env = kwargs.get('env', 'fourfront-mastertest')  # by default
+    branch = kwargs.get('branch', 'master')  # by default deploy master
+    application_version_name = kwargs.get('application_version_name', None)
+    repo = kwargs.get('repo', None)
+
+    if application_version_name is None:  # if not specified, use branch+timestamp
+        application_version_name = 'foursight-package-%s-%s' % (branch, datetime.datetime.utcnow())
+
+    if repo is not None:
+        repo_location = clone_repo_to_temporary_dir(repo)
+    else:
+        repo_location = clone_repo_to_temporary_dir()
+
+    packaging_was_successful = EBDeployer.build_application_version(repo_location,
+                                                                    application_version_name,
+                                                                    branch=branch)
+    time.sleep(10)  # give EB some time to index the new template
+    if packaging_was_successful:
+        try:
+            EBDeployer.deploy_new_version(env, repo_location, application_version_name)
+            check.status = 'PASS'
+            check.summary = 'Successfully deployed version %s to env %s' % (application_version_name, env)
+        except Exception as e:
+            check.status = 'ERROR'
+            check.summary = 'Error encountered while deploying: %s' % str(e)
+    else:
+        check.status = 'ERROR'
+        check.summary = 'Could not package repository: %s' % packaging_was_successful
```

Comment:
If we got here, packaging_was_successful will just have None or the empty string, right? I guess it can't hurt to show but it's a pretty boring value. Is there no error message somewhere?

Reply:
It depends what happens in `build_application_version`. It could raise an error, return False, or return the response to the boto3 API call: `create_application_version`. Each one informs a different situation at least, but probably not an ideal API to have on my part.